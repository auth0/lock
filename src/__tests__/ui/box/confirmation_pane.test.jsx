import React from 'react';
import I from 'immutable';

import { expectComponent } from 'testUtils';

import ConfirmationPane from '../../../ui/box/confirmation_pane';

const defaultProps = {
  lock: I.fromJS({ id: '__lock-id__' }),
  backHandler: () => {},
  closeHandler: () => {},
  children: <span>test</span>,
  svg: <svg>svg</svg>
};

describe('ConfirmationPane', () => {
  it('renders correctly', async () => {
    expectComponent(<ConfirmationPane {...defaultProps} />).toMatchSnapshot();
  });
});
