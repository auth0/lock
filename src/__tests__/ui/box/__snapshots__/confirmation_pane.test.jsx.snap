// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPane renders correctly 1`] = `
<div
  className="auth0-lock-confirmation"
>
  <span
    aria-label="close"
    className="auth0-lock-close-button"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" enable-background=\\"new 0 0 128 128\\" version=\\"1.1\\" viewBox=\\"0 0 128 128\\" xml:space=\\"preserve\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><g><polygon fill=\\"#373737\\" points=\\"123.5429688,11.59375 116.4765625,4.5185547 64.0019531,56.9306641 11.5595703,4.4882813     4.4882813,11.5595703 56.9272461,63.9970703 4.4570313,116.4052734 11.5244141,123.4814453 63.9985352,71.0683594     116.4423828,123.5117188 123.5126953,116.4414063 71.0732422,64.0019531   \\"></polygon></g></svg>",
      }
    }
    id="__lock-id__-close-button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  />
  <span
    aria-label="back"
    className="auth0-lock-back-button"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" enable-background=\\"new 0 0 24 24\\" version=\\"1.0\\" viewBox=\\"0 0 24 24\\" xml:space=\\"preserve\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"> <polyline fill=\\"none\\" points=\\"12.5,21 3.5,12 12.5,3 \\" stroke=\\"#000000\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"></polyline> <line fill=\\"none\\" stroke=\\"#000000\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\" x1=\\"22\\" x2=\\"3.5\\" y1=\\"12\\" y2=\\"12\\"></line> </svg>",
      }
    }
    id="__lock-id__-back-button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  />
  <div
    className="auth0-lock-confirmation-content"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": <svg>
            svg
          </svg>,
        }
      }
    />
    <span>
      test
    </span>
  </div>
</div>
`;
